File: backend/routes/security.py
from fastapi import APIRouter

from services.security_service import SecurityService


router = APIRouter(
    prefix="/security",
    tags=["Security"]
)

security_service = SecurityService()


# =========================================================
# SECURITY ANALYSIS
# =========================================================

@router.post("/{owner}/{repo}")
def analyze_security(
    owner: str,
    repo: str
):
    try:

        result = security_service.analyze_security(
            owner,
            repo
        )

        return result

    except Exception as e:

        return {
            "error": str(e)
        }

File: frontend/src/services/security.ts
const API = "http://127.0.0.1:8000";

export async function analyzeSecurity(
  owner: string,
  repo: string
) {
  const response = await fetch(
    `${API}/security/${owner}/${repo}`,
    {
      method: "POST",
    }
  );

  if (!response.ok) {
    throw new Error("Failed to analyze security");
  }

  return await response.json();
}